### PR TITLE
Compile to tplan

### PR DESCRIPTION
**Pull request by wafle**

### CODE REVIEW

File: compiler/src/test/scala/ingraph/sandbox/JPlanToFPlanTest.scala [hunk 1]
```diff
@@ -73,128 +97,124 @@ class SchemaInferencerTest extends FunSuite {
...
       """MATCH (a:A)-[:R1]->(b:B)-[:R2]->(c:C)
         |RETURN a, b, c
         |""".stripMargin)
-    //    println(ep.inode)
-    //    println(ep.children(0).internalSchema)
-    //    println(ep.children(0).children(0).internalSchema)
-    //    println(ep.children(0).children(0).children(0).internalSchema)
-    //    println(ep.children(0).children(0).children(0).children(0).internalSchema)
-    //    println(ep.children(0).children(0).children(0).children(1).internalSchema)
+    // TODO: assert
```
Comment (wafle) [inline compiler/src/test/scala/ingraph/sandbox/JPlanToFPlanTest.scala]: :)

File: compiler/src/test/scala/ingraph/sandbox/JPlanToFPlanTest.scala [hunk 2]
```diff
@@ -209,19 +229,19 @@ class SchemaInferencerTest extends FunSuite {
         |  (segment6:Segment)-[mb6:monitoredBy]->(sensor:Sensor)
         |RETURN sensor, segment1, segment2, segment3, segment4, segment5, segment6
         |""".stripMargin)
-//    println(ep)
-//    println(ep.children(0).children(0).children(0).internalSchema)
-//    println(ep.children(0).children(0).children(0).asInstanceOf[Join].leftMask)
-//    println(ep.children(0).children(0).children(0).asInstanceOf[Join].rightMask)
+    // TODO: assert
```
Comment (wafle) [inline compiler/src/test/scala/ingraph/sandbox/JPlanToFPlanTest.scala]: :)

File: ire-adapter/src/main/scala/ingraph/ire/EngineFactory.scala [hunk 3]
```diff
@@ -70,14 +70,14 @@ object EngineFactory {
...
+          case op: BinaryTNode =>
             val node: ActorRef = op match {
-              case op: Union => newLocal(Props(new UnionNode(expr.child, op.jnode.bag)))
+              case op: Union => newLocal(Props(new UnionNode(expr.child, op.fnode.jnode.bag)))
               case op: JoinLike =>
                 val child = expr.child
-                val leftTupleWidth = op.left.internalSchema.length
-                val rightTupleWidth = op.right.internalSchema.length
-                val leftMask = op.leftMask
+                val leftTupleWidth  = op.left. fnode.internalSchema.length
```
Comment (wafle) [inline ire-adapter/src/main/scala/ingraph/ire/EngineFactory.scala]: whats up with the whitespace here?

Comment (szarnyasg) [reply ire-adapter/src/main/scala/ingraph/ire/EngineFactory.scala]: well, left and right are of different lengths :-)
I even considered moving to `pri` and `sec`, but let's leave that to the Rete network

**Pull request merged**